backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interest/InterestService.java

```diff
@@ -66,4 +75,30 @@ public void distributeInterestAndBonus(@NonNull final InterestBonusComputationRe
 				.build()
 				.distributeInterestAndBonus(request);
 	}
+
+	public InterestBonusComputationRequest getInterestBonusComputationRequest(final EnqueueInterestComputationRequest enqueueRequest)
```

Comment:
method is called `get..` but inside we are locking some logs.

* pls make it more intention revealing
* where the lock is released?

Reply:
The method is about creating the request object (which integrates a lock on )
`de.metas.contracts.modular.interest.InterestComputationWorkPackageProcessor#processWorkPackage`
and `de.metas.cucumber.stepdefs.contract.ModCntr_Interest_Run_StepDef#distributeInterest` respectively, where the InterestBonusComputationRequest is consumed.

The field in InterestBonusComputationRequest is called `involvedModularLogsLock` so, to me, it looks as intention revealing as possible. I've extracted a private method lockModularContractLogsForInvoicingGroup(...) to make it even clearer that a lock is acquired at this point.